app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cash_result = {}
+
+    def wrapper(*args) -> Any:
```

Comment:
what if func takes key arguments?
don't forget about kwargs

Reply:
Also, What if two different func takes the same arguments?

Reply:
> Also, What if two different func takes the same arguments?

Will return results for each function separately